fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -621,8 +621,10 @@ ACTOR Future<Void> shardTracker(
 			wait( delay(0, TaskPriority::DataDistribution) );
 		}
 	} catch (Error& e) {
-		if (e.code() != error_code_actor_cancelled)
-			self->output.sendError(e);		// Propagate failure to dataDistributionTracker
+		// If e is broken_promise then self may have already been deleted
+		if (e.code() != error_code_actor_cancelled && e.code() != error_code_broken_promise) {
+			self->output.sendError(e); // Propagate failure to dataDistributionTracker
```

Comment:
I didn't see any bad effect for this. 
What I'm puzzled at is: When will broken_promise be thrown from the actor?
It seems the `shardMerger` has handles it https://github.com/xumengpanda/foundationdb/blob/4086e3a2750b776cc8bfb0f0e463fe00ac905595/fdbserver/DataDistributionTracker.actor.cpp#L471 

Maybe  `shardSplitter` may throw out broken_promise error? -- I'm just guessing without digging further.

Reply:
The broken_promise comes from `DataDistributionTracker::anyZeroHealthyTeams` getting destroyed. `shardEvaluator` waits on `anyZeroHealthyTeams->onChange()` which is how the error gets here.